site/src/comparison.rs

```diff
@@ -180,34 +185,39 @@ async fn populate_report(
     }
 }
 
-#[derive(Copy, Clone, Debug, PartialEq, Serialize, Deserialize)]
+#[derive(Clone, Debug, PartialEq, Serialize)]
 pub enum Metric {
     #[serde(rename = "instructions:u")]
     Instructions,
     #[serde(rename = "cycles:u")]
     Cycles,
-    #[serde(rename = "faults")]
-    Faults,
     #[serde(rename = "max-rss")]
     MaxRSS,
-    #[serde(rename = "task-clock")]
-    TaskClock,
-    #[serde(rename = "wall-time")]
-    WallTime,
-    #[serde(rename = "cpu-clock")]
-    CpuClock,
+    Custom(String),
```

Comment:
I think if we do this, I would like a comment or perhaps even a wrapper around the String in a private module that (tries to) guarantee we're not going to end up with `.to_string() == "size:some_custom_size"` down the line. I'd really rather avoid string-y typing for these metrics if we can, and it's not clear to me that an unbounded set of metrics is a good idea. If we can't justify the set of metrics enough to *name them*, then maybe they shouldn't be collected -- after all, we do pay some amount of overhead the larger the set is (in mental complexity at least, though infrastructure cost too).

Reply:
I wanted to create explicit enum variants for enums used explicitly in code (currently cycles, instructions and max RSS) and use an opaque string wrapper for the rest. I use a string wrapper, because the metrics are coming from "unbounded" sources at this moment:
- Artifact sizes are read from a self-profile JSON, which can basically contain arbitrary data, as far as `rustc-perf` is concerned.
- Perf. metrics (with the exception of wall time, cycles, instructions and cpu-clock) are also read as an arbitrary string from the output of `perf`.

Things are also a bit more complicated by the fact that some metrics exist in two very similar variants (e.g. `cycles` and `cycles:u`), and even if we wanted to unify these somehow, they are stored in historical data, so we probably need to support them.

So I see three ways of resolving this:
- Use the proposed way of using an opaque wrapper for unknown metrics.
- Make a list of known metrics and ignore the rest.
- Make a list of known metrics and crash if an unknown one is encountered (in that case we should add it to the `Metric` enum).